### PR TITLE
diff merged all I could find about modules for 5.5

### PR DESCRIPTION
**Pull request by guyboertje**
TODO:

- [x] Add the 5.5 specific bits

### CODE REVIEW

File: logstash-core/lib/logstash/elasticsearch_client.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      @settings = settings
+      @logger = logger
+      @client = Elasticsearch::Client.new(client_args)
+    end
+
+    def delete(path)
+      begin
+        normalize_response(@client.perform_request('DELETE', path, {}, nil))
+      rescue Exception => e
+        if e.class.to_s =~ /NotFound/ || e.message =~ /Not\s*Found|404/i
```
Comment (ph) [inline logstash-core/lib/logstash/elasticsearch_client.rb]: is_404?

File: logstash-core/lib/logstash/runner.rb [hunk 2]
```diff
@@ -243,6 +274,15 @@ def execute
 
     @settings.format_settings.each {|line| logger.debug(line) }
 
+    module_configs = LogStash::Config::ModulesCommon.pipeline_configs(@settings)
+    module_config_hash = module_configs.first
+    if !module_config_hash.nil?
+      @settings.set_value("config.string", config_hash["config_string"])
```
Comment (ph) [inline logstash-core/lib/logstash/runner.rb]: I think it should be `module_config_hash` instead of config_hash?

Comment (guyboertje) [inline logstash-core/lib/logstash/runner.rb]: true

File: logstash-core/lib/logstash/config/modules_common.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          alt_name = "module-#{module_hash["name"]}"
+          pipeline_id = alt_name
+
+          current_module.with_settings(module_hash)
+          current_module.import(import_engine)
+          config_string = current_module.config_string
+
+          pipelines << {"pipeline_id" => pipeline_id, "alt_name" => alt_name, "config_string" => config_string, "settings" => settings}
+        rescue => e
+          raise LogStash::ConfigLoadingError, I18n.t("logstash.modules.configuration.parse-failed", :error => e.message)
```
Comment (ph) [inline logstash-core/lib/logstash/config/modules_common.rb]: This custom exception doesn't exist in 5.x.

```
#<NameError: uninitialized constant LogStash::ConfigLoadingError>,
```

Comment (guyboertje) [inline logstash-core/lib/logstash/config/modules_common.rb]: hehe I saw that right this minute myself when I tried it.

Comment (ph) [inline logstash-core/lib/logstash/config/modules_common.rb]:
```diff
diff --git a/logstash-core/lib/logstash/errors.rb b/logstash-core/lib/logstash/errors.rb
index 8960a7f12..e27247ad8 100644
--- a/logstash-core/lib/logstash/errors.rb
+++ b/logstash-core/lib/logstash/errors.rb
@@ -6,6 +6,7 @@ module LogStash
   class PluginLoadingError < Error; end
   class ShutdownSignal < StandardError; end
   class PluginNoVersionError < Error; end
+  class ConfigLoadingError < Error; end
 
   class Bug < Error; end
   class ThisMethodWasRemoved < Bug; end

Comment (ph) [inline logstash-core/lib/logstash/config/modules_common.rb]: Starting es, with the previous diff make it work:

```

15:51:23,286][WARN ][logstash.outputs.elasticsearch] Restored connection to ES instance {:url=>#<Java::JavaNet::URI:0x6f3b13b9>}
[2017-06-06T15:51:23,298][INFO ][logstash.outputs.elasticsearch] New Elasticsearch output {:class=>"LogStash::Outputs::ElasticSearch", :hosts=>[#<Java::JavaNet::URI:0x39ae9fd0>]}
[2017-06-06T15:51:23,301][INFO ][logstash.filters.geoip   ] Using geoip database {:path=>"/Users/ph/es/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-geoip-4.1.1-java/vendor/GeoLite2-City.mmdb"}
[2017-06-06T15:51:23,337][INFO ][logstash.filters.geoip   ] Using geoip database {:path=>"/Users/ph/es/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-geoip-4.1.1-java/vendor/GeoLite2-City.mmdb"}
[2017-06-06T15:51:23,338][INFO ][logstash.filters.geoip   ] Using geoip database {:path=>"/Users/ph/es/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-geoip-4.1.1-java/vendor/GeoLite2-City.mmdb"}
[2017-06-06T15:51:23,341][INFO ][logstash.pipeline        ] Starting pipeline {"id"=>"main", "pipeline.workers"=>4, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>5, "pipeline.max_inflight"=>500}
[2017-06-06T15:51:23,357][INFO ][logstash.inputs.tcp      ] Starting tcp input listener {:address=>"0.0.0.0:5000"}
[2017-06-06T15:51:23,373][INFO ][logstash.pipeline        ] Pipeline main started
[2017-06-06T15:51:23,444][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
```

**Pull request merged**